ddtrace/llmobs/_experiment.py

```diff
@@ -161,6 +159,50 @@ def __len__(self) -> int:
     def __iter__(self) -> Iterator[DatasetRecord]:
         return iter(self._records)
 
+    def as_dataframe(self):
+        try:
+            import pandas as pd
+        except ImportError as e:
+            raise ImportError(
+                "pandas is required to convert dataset to DataFrame. " "Please install it with `pip install pandas`"
+            ) from e
+
+        column_tuples = set()
+        data_rows = []
+        for record in self._records:
+            flat_record = {}
+
+            input_data = record.get("input_data", {})
+            if isinstance(input_data, dict):
+                for k, v in input_data.items():
+                    flat_record[("input_data", k)] = v
+                    column_tuples.add(("input_data", k))
+            else:
+                flat_record[("input_data", "")] = input_data  # Use empty string for single input
+                column_tuples.add(("input_data", ""))
+
+            expected_output = record.get("expected_output", {})
+            if isinstance(expected_output, dict):
+                for k, v in expected_output.items():
+                    flat_record[("expected_output", k)] = v
+                    column_tuples.add(("expected_output", k))
+            else:
+                flat_record[("expected_output", "")] = expected_output  # Use empty string for single output
+                column_tuples.add(("expected_output", ""))
+
+            for k, v in record.get("metadata", {}):
+                flat_record[("metadata", k)] = v
+                column_tuples.add(("metadata", k))
```

Comment:
This might be a nit, but we don't actually do anything with the record metadata field atm (not part of the eval metrics or span submitted to experiments). Ambivalent about keeping it or removing it until we need it, but will defer to your judgement